### PR TITLE
Add nf-test for bcl_demultiplex, remove old pytest

### LINKED ISSUE #4495: [FEATURE] Add nf-test to subworkflow bcl_demultiplex
### Is your feature request related to a problem? Please describe

Migrate the pytest to nf-tests for the bcl_demultiplex subworkflow

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by karthikram-takara**
## PR checklist

Closes #4495  

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: subworkflows/nf-core/bcl_demultiplex/tests/main.nf.test.snap [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+    "reports": {
+        "content": [
+            [
```
Comment (LeuThrAsp) [inline subworkflows/nf-core/bcl_demultiplex/tests/main.nf.test.snap:4]: no test files

**Review by LeuThrAsp — CHANGES_REQUESTED** [review]
snap file does not include files

File: subworkflows/nf-core/bcl_demultiplex/tests/main.nf.test.snap (hunk 1 above)
Comment (karthikram-takara) [reply subworkflows/nf-core/bcl_demultiplex/tests/main.nf.test.snap:4]: Thanks, fixed this in later commits.

**Comment by SPPearce** [comment]
Closing as superceded by #4842

**Pull request closed without merging**